lib/buffer-patch-helpers.coffee

```diff
@@ -0,0 +1,22 @@
+composeChangedLines = (editor, {oldRange, oldText, newRange, newText}) ->
+  startRow = Math.min(oldRange.start.row, newRange.start.row)
+  endRow = Math.max(oldRange.end.row, newRange.end.row)
+
+  newRangeStartIndex = linearIndexInRowRange(editor, newRange.start, startRow, endRow)
+  newRangeEndIndex = linearIndexInRowRange(editor, newRange.end, startRow, endRow)
+
+  newLines = editor.getTextInBufferRange([[startRow, 0], [endRow, 10000]])
```

Comment:
Yeah I'll remove that thanks for the reminder. I wanted to get the entire line thus the huge number. I could check the length of the row. 

I wanted to get the whole line in case the range end was mid-word, would not want to add half a word.